src/models/application.ts

```diff
@@ -455,40 +450,39 @@ const getApplicationModel = function (
 		 * 	console.log(device);
 		 * })
 		 */
-		async getWithDeviceServiceDetails(
-			slugOrUuidOrId: string | number,
-			options?: PineOptions<Application>,
-		): Promise<
-			Application & {
-				owns__device: Array<DeviceWithServiceDetails<CurrentServiceWithCommit>>;
-			}
-		> {
-			options ??= {};
-			const serviceOptions = mergePineOptions(
-				{
-					$expand: [
-						{
-							owns__device: {
-								$expand: getCurrentServiceDetailsPineExpand(true),
-							},
-						},
-					],
+		async getWithDeviceServiceDetails<
+			T extends ODataOptionsWithoutCount<Application['Read']>,
+		>(slugOrUuidOrId: string | number, options?: T) {
+			const baseOptions = {
+				$expand: {
+					owns__device: {
+						$expand: getCurrentServiceDetailsPineExpand,
+					},
 				},
+			} as const;
+
+			const serviceOptions: typeof baseOptions = mergePineOptions(
```

Comment:
This is needed as the current `MergePineOptions` cannot infer while the generic T is not defined that the return type will indeed have all the properties needed to resolve the query, but it knows that it satisfies so we can simply declare the typing so it works withing function scope